### PR TITLE
Request to merge Week 3 Homework assignment

### PR DESCRIPTION
**Pull request by rors**

### CODE REVIEW

**Comment by rors** [comment]
@alexhsamuel want to review this?

File: homeworks/week03/README.md [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+will start with a file of static HTML. The HTML markup in the file is
+very similar to what you generated last week, but there are some
+differences, and working with a static HTML file that you can examine
+in your text editor will hopefully make things easier than using the
+JavaScript `render()` functions.
+
+The goal will be to handle various events, so that what is displayed
+is dynamically changed by user actions.
+
+As with the previous homework, to begin, download the zip file for
```
Comment (alexhsamuel) [inline homeworks/week03/README.md:21]: Do we really need a zip file? There are just a few files, which will be in the github repo, right?

File: homeworks/week03/README.md [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+movie only if `movie_year` is greater than `user_year` *and* `checked`
+is `true`.
+
+Once you get that worked out, your genre filters and year filters
+should be working together.
+
+## Bonus
+
+If you've gotten this far and still have time, try adding another form
+that let's the user input the title, genre, year, description and
```
Comment (alexhsamuel) [inline homeworks/week03/README.md]: let's -> lets

File: homeworks/week03/README.md [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+
+Replace "___" by using a selector to target your year `<input>` and
+calling `.val()` on that. We've wrapped it in `parseInt()` because
+`.val()` returns a string, and we want an actual numerical value that
+we can do a comparison on.
+
+Now add the following:
+
+```javascript
+  $('div.movie').each(function() {
```
Comment (alexhsamuel) [inline homeworks/week03/README.md]: Space between `function` and `(` for anonymous functions.

File: homeworks/week03/README.md [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+handler. First, comment out your solutions to Steps 2 and 3. Don't 
+delete them, because we'll refer back to them. But we don't need that
+functionality now because we will be recreating it in a better way 
+here. But don't comment out the `preventDefault()`! That is still
+needed.
+
+Start with the following event handler:
+
+```javascript
+  $('input').bind('keyup change', function(event) {
```
Comment (alexhsamuel) [inline homeworks/week03/README.md]: Space between `function` and `(`.

File: homeworks/week03/README.md [hunk 5]
```diff
@@ -0,0 +1,266 @@
...
+events for each `input` on the page.
+
+Inside that, copy and paste your line from Step 3 that sets `var
+user_year` to the value of your year `<input>`.
+
+Add the following `if` to make sure we have a valid value: (Question:
+Can you think of why we didn't need to worry about this in Step 3?)
+
+```javascript
+  if ( isNaN(user_year) ) {
```
Comment (alexhsamuel) [inline homeworks/week03/README.md]: No spaces inside ().

File: homeworks/week03/README.md [hunk 6]
```diff
@@ -0,0 +1,266 @@
...
+
+Add a [keyup event handler](https://api.jquery.com/keyup/) there to
+respond whenever the user enters a character into the text box. You
+can target the text box input by its ID.
+
+Inside this handler, let's start by getting the numerical value of the
+year that the user has just entered. Start with this:
+
+```javascript
+  var user_year = parseInt( ___ )
```
Comment (alexhsamuel) [inline homeworks/week03/README.md]: `user_year` -> `userYear`.

File: homeworks/week03/README.md [hunk 7]
```diff
@@ -0,0 +1,266 @@
...
+calling `.val()` on that. We've wrapped it in `parseInt()` because
+`.val()` returns a string, and we want an actual numerical value that
+we can do a comparison on.
+
+Now add the following:
+
+```javascript
+  $('div.movie').each(function() {
+
+      var movie_year = parseInt( $(this).find('dd.year').html() )
```
Comment (alexhsamuel) [inline homeworks/week03/README.md]: `movie_year` -> `movieYear`.

File: homeworks/week03/README.md (hunk 4 above)
Comment (alexhsamuel) [inline homeworks/week03/README.md]: Space after `function`.

**Comment by alexhsamuel** [comment]
I think the assignment is reasonable, but the instructions are hard to follow. I'm not sure I have any great suggestions though. Part of me wonders whether it would be clearer if the instructions were _less_ explicit.

One thing you might try is to give simpler functional specifications, and break out some hints at the end that provide more details for how the logic might be structured.

File: homeworks/week03/README.md (hunk 2 above)
Comment (rors) [inline homeworks/week03/README.md]: thx for the catch

File: homeworks/week03/README.md (hunk 3 above)
Comment (rors) [inline homeworks/week03/README.md]: changed.

File: homeworks/week03/README.md (hunk 5 above)
Comment (rors) [inline homeworks/week03/README.md]: changed.

File: homeworks/week03/README.md (hunk 4 above)
Comment (rors) [inline homeworks/week03/README.md]: changed.

File: homeworks/week03/README.md (hunk 6 above)
Comment (rors) [inline homeworks/week03/README.md]: changed.

File: homeworks/week03/README.md (hunk 7 above)
Comment (rors) [inline homeworks/week03/README.md]: change.d

**Comment by rors** [comment]
I agree that perhaps less explicit would be better. But I was struggling with how to strike a balance on how to write instructions for the declarative jQuery style. Let's talk more about this offline.

**Comment by rors** [comment]
Otherwise, all above changes have been made and pushed. I think this is ready to merge. @alexhsamuel @ckuang @nmadd, anyone want to marge? or else I can go ahead and do it. Thx
